src/LevelParser.cpp

```diff
@@ -0,0 +1,186 @@
+#include "LevelParser.h"
+#include "XML\tinystr.h"
+#include "XML\tinyxml.h"
+#include "WorldMap.h"
+#include "Base64.h"
+#include <assert.h>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include <vector>
+#include <SFML\Graphics.hpp>
+#include <utility>
+
+struct LevelDetails
+{
+	LevelDetails(const int tileSize, const sf::Vector2i& mapSize)
+		: m_tileSize(tileSize),
+		m_mapSize(mapSize)
+	{}
+
+	const int m_tileSize;
+	const sf::Vector2i m_mapSize;
+};
+
+void parseTileMap(WorldMap& worldMap, const TiXmlElement & root, const LevelDetails & levelDetails);
+void parseTileLayer(WorldMap& worldMap, const TiXmlElement & tileLayerElement, const LevelDetails& levelDetails, std::string&& tileLayerName);
+LevelDetails parseLevelDetails(const TiXmlElement& root);
+std::vector<std::vector<int>> decodeTileLayer(const TiXmlElement & tileLayerElement, const LevelDetails & levelDetails);
+const TiXmlElement* findNode(const TiXmlElement& root, const std::string& name);
+const TiXmlElement* findNode(const TiXmlElement& root, const std::string& value, const std::string& name);
+int getNumberOfTileLayers(const TiXmlElement& root);
+int getNumberOfTileSets(const TiXmlElement& root);
+
+void LevelParser::parseLevel(WorldMap& worldMap, const std::string & levelName)
+{
+	//Load XML file
+	TiXmlDocument levelDocument;
+	const bool isOpen = levelDocument.LoadFile(levelName.c_str());
+	assert(isOpen);
+
+	const TiXmlElement& rootNode = *levelDocument.RootElement();
+	const LevelDetails levelDetails = parseLevelDetails(rootNode);
+	parseTileMap(worldMap, rootNode, levelDetails);
+}
+
+void parseTileLayer(WorldMap& worldMap, const TiXmlElement & tileLayerElement, const LevelDetails& levelDetails, std::string&& tileLayerName)
+{
+	worldMap.assignTileLayer(decodeTileLayer(tileLayerElement, levelDetails), levelDetails.m_mapSize, std::move(tileLayerName));
+}
+
+LevelDetails parseLevelDetails(const TiXmlElement & root)
+{
+	int width = 0, height = 0, tileSize = 0;
+	root.Attribute("width", &width);
+	root.Attribute("height", &height);
+	root.Attribute("tilewidth", &tileSize);
```

Comment:
parse tileheight, maybe assert equal to width or use width/height everywhere instead of tileSize